### PR TITLE
docs: add docs for BSC staking and gov

### PR DESCRIPTION
**Pull request by forcodedancing**
do not need to merge now

it is preparation

### CODE REVIEW

File: docs/bcfusion/validators/staking.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
+# Staking
+
+BNB smart chain (BSC) is a Proof-of-Staked-Authority (PoSA) blockchain, which means that staking is one of the most
+important parts of the system. [BEP-294](https://github.com/bnb-chain/BEPs/pull/294) introduces the new native staking
+mechanism after BNB chain fusion, which has several improvements:
+
+- Users are able to participate in staking on the BSC using BNB in BEP20 format.
+- Staking credit, as the proof of BNB staked to a specified validator, cannot be transferred. Different validators issue
+  different staking credits.
+- The rewards for staking can be claimed more quickly.
```
Comment (Pythonberg1997) [inline docs/bcfusion/validators/staking.md:10]: The rewards don't need to be claimed

File: docs/bcfusion/validators/staking.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+To create a validator, a BNB holder needs to send a `CreateValidator` transaction to the BNB smart chain,
+with minimum amount of BNB that the validator needs to stake to their own validator
+address (2000 BNB), specifying the following information:
+
+- **Operator address**: The address of the validator, which will receive the staking credit and the rewards.
+- **Consensus address**: The consensus address of the validator's node.
+- **Vote Address**: The address for participating fast finality voting.
+- **BLS Proof**: The data to prove that the validator owns the vote address.
+- **Commission rate**: The percentage of the rewards that the validator will keep for themselves, and the rest will be
```
Comment (Pythonberg1997) [inline docs/bcfusion/validators/staking.md:44]: The commission info has three params: commissionRate, maxRate, maxChangeRate

File: docs/bcfusion/validators/staking.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+- **Amount**: The amount of BNB that the delegator wants to unstake from the validator.
+
+The `Undelegate` transaction will deduct the amount of staking credit from the validator and return the corresponding
+BNB to the delegator address. However, the BNB will be locked for a period of time, called the **unbonding period**,
+before the delegator can use it. The unbonding period is currently set to 7 days, and it is designed to prevent
+delegators from quickly withdrawing their BNB in case of a validator misbehavior or a network attack.
+
+### Claim
+
+To claim the unbond BNB and the rewards, a delegator should send a `Claim` transaction to the BNB smar chain, specifying
```
Comment (Pythonberg1997) [inline docs/bcfusion/validators/staking.md:124]: `BNB smart chain`

File: docs/bcfusion/validators/gov.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+- **Proposer address**: The BEP20 address of the proposer, who initiates the proposal and pays the proposal fee.
+- **Targets**: The list of addresses of the contracts or accounts that the proposal wants to interact with.
+- **Values**: The list of amounts of BNB or other tokens that the proposal wants to transfer to the targets.
+- **Signatures**: The list of function signatures of the contracts that the proposal wants to call on the targets.
+- **Calldatas**: The list of encoded arguments of the functions that the proposal wants to call on the targets.
+- **Description**: The description of the proposal, which provides more details and rationale for the proposal.
+
+A delegator should delegate more the 200 staked BNB as the minimal requirement for submitting a proposal.
+The `propose` transaction will deduct the proposal fee (TODO: currently there is no such fee, we may add the fee)
```
Comment (zjubfd) [inline docs/bcfusion/validators/gov.md:32]: let us remove the fee part here first.

File: docs/bcfusion/validators/gov.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+from the proposer address and create a new proposal on the BNB
+smart chain. The proposal will have a unique proposal ID, and a proposal status of `Pending`. The proposal will then
+enter the voting period, which is the time window for the staking credit holders to cast their votes on the proposal.
+
+### Cast Vote
+
+To cast a vote on a proposal, a staking credit holder needs to send a `castVote` transaction to the BNB smart chain,
+specifying the following information:
+
+- **Voter address**: The BEP20 address of the voter, who casts the vote on the proposal. The voter address can be the
```
Comment (zjubfd) [inline docs/bcfusion/validators/gov.md:42]: There is no concept called `BEP20 address`

File: docs/bcfusion/validators/gov.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+The `castVote` transaction will record the support value and the voting power of the voter on the BNB smart chain. The
+voting power is the amount of staking credit that the voter has at the time of the vote. The voting power can change due
+to staking operations, such as delegate, undelegate, or redelegate, but the support value will remain the same. The
+voter can change their support value at any time during the voting period, by sending another `castVote` transaction
+with a different support value.
+
+After submitting a proposal, the staking credit holders can cast their votes on the proposal until the voting period
+ends. The voting period is one of the BNB smart chain governance parameters, and it is currently set to 7 days.
+
```
Comment (zjubfd) [inline docs/bcfusion/validators/gov.md:57]: can we add a part about how we tally the votes, in what condition the proposal is approved?

File: docs/bcfusion/validators/gov.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+In addition to casting votes, the staking credit holders can also delegate their voting power to someone else to
+participate in governance on their behalf. This can be useful for staking credit holders who do not have the time,
+interest, or expertise to follow and vote on the proposals, but still want to have a say in the governance process. By
+delegating their voting power to a trusted party, such as a validator, a friend, or a professional service, they can
+benefit from their knowledge and experience, and also avoid the risk of losing their staking rewards due to abstaining
+from voting.
+
+To delegate their voting power, a staking credit holder needs to send a `delegateVote` transaction to the BNB
+smart chain, specifying the following information:
```
Comment (zjubfd) [inline docs/bcfusion/validators/gov.md:92]: invoke which contract?

File: docs/bcfusion/validators/gov.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+to staking operations, such as delegate, undelegate, or redelegate, but the support value will remain the same. The
+voter can change their support value at any time during the voting period, by sending another `castVote` transaction
+with a different support value.
+
+After submitting a proposal, the staking credit holders can cast their votes on the proposal until the voting period
+ends. The voting period is one of the BNB smart chain governance parameters, and it is currently set to 7 days.
+
+### Execute Proposal
+
+To execute a proposal, anyone can send an `execute` transaction to the BNB smart chain, specifying the following
```
Comment (zjubfd) [inline docs/bcfusion/validators/gov.md:60]: interact with which contract?

File: docs/bcfusion/validators/staking.md [hunk 9]
```diff
@@ -0,0 +1,133 @@
+# Staking
+
+BNB smart chain (BSC) is a Proof-of-Staked-Authority (PoSA) blockchain, which means that staking is one of the most
+important parts of the system. [BEP-294](https://github.com/bnb-chain/BEPs/pull/294) introduces the new native staking
+mechanism after BNB chain fusion, which has several improvements:
+
+- Users are able to participate in staking on the BSC using BNB in BEP20 format.
```
Comment (zjubfd) [inline docs/bcfusion/validators/staking.md:7]: I think BNB is a native token, not a BEP20 format.

File: docs/bcfusion/validators/staking.md [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+- Staking credit, as the proof of BNB staked to a specified validator, cannot be transferred. Different validators issue
```
Comment (zjubfd) [inline docs/bcfusion/validators/staking.md:8]: Is this an improvement?

File: docs/bcfusion/validators/staking.md [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+  different staking credits.
+- The rewards for staking can be claimed more quickly.
+
+In this section, we will explain the basic concepts and operations of staking on the BSC.
+
+## Validator Set
+
+The validator set is the group of nodes that are responsible for validating transactions and producing blocks on the
+BSC. The validator set is determined by the amount of staking credit each validator has, which reflects the amount of
+BNB staked by the validator and its delegators. The top 45 validators with the most staking credit are selected as the
```
Comment (zjubfd) [inline docs/bcfusion/validators/staking.md:18]: `45` is a temp value, will grow to 100 and more in the future

File: docs/bcfusion/validators/staking.md [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+## Validator Operations
+
+Validators are the nodes that run the BNB smart chain software and participate in the consensus process. Validators need
+to have a minimum amount of BNB staked to their own validator address, and they can also accept delegations from other
+BNB holders who want to stake with them. Validators earn rewards from the transaction fees, and
+they share a portion of their rewards with their delegators.
+
+### Create Validator
+
+To create a validator, a BNB holder needs to send a `CreateValidator` transaction to the BNB smart chain,
```
Comment (zjubfd) [inline docs/bcfusion/validators/staking.md:36]: `send a `xxx` transaction to the BNB smart chain`  this is a too general speaking,  to which contract is better.

File: docs/bcfusion/validators/staking.md [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+with minimum amount of BNB that the validator needs to stake to their own validator
+address (2000 BNB), specifying the following information:
+
+- **Operator address**: The address of the validator, which will receive the staking credit and the rewards.
```
Comment (zjubfd) [inline docs/bcfusion/validators/staking.md:40]: and commission

File: docs/bcfusion/validators/staking.md [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+- **Consensus address**: The consensus address of the validator's node.
+- **Vote Address**: The address for participating fast finality voting.
+- **BLS Proof**: The data to prove that the validator owns the vote address.
```
Comment (zjubfd) [inline docs/bcfusion/validators/staking.md:43]: `The data` ==> A BLS signature

File: docs/bcfusion/validators/staking.md [hunk 15]
```diff
@@ -0,0 +1,133 @@
...
+  validator.
+- **Delegate Voting Power**: The flag to indicate whether the delegator would like to delegate his/her voting power
+  to the destination validator for governance.
+
+The `Redelegate` transaction will deduct the amount of staking credit from the source validator and issue the
+corresponding staking credit to the destination validator. The delegator will then share the rewards and the risks with
+the destination validator, according to the commission rate and the slashing rate of the destination validator.
+The `Redelegate` transaction does not incur the unbonding period, but it will incur the redelegation fee,
+which is designed to prevent delegators from frequently switching between validators to chase
+the highest rewards or avoid the highest risks.
```
Comment (zjubfd) [inline docs/bcfusion/validators/staking.md:107]: Please add the default fee rate here.

File: docs/bcfusion/validators/staking.md [hunk 16]
```diff
@@ -0,0 +1,133 @@
...
+
+### Undelegate
```
Comment (zjubfd) [inline docs/bcfusion/validators/staking.md:109]: can we explain the core logic about how user earn reward from staking..
I mean how to turn `share` to real `BNB` reward, the key formula?

File: docs/bcfusion/validators/staking.md [hunk 17]
```diff
@@ -0,0 +1,133 @@
...
+
+To claim the unbond BNB and the rewards, a delegator should send a `Claim` transaction to the BNB smar chain, specifying
+the following information:
+
+- **Delegator address**: The BEP20 address of the delegator, which will receive the rewards from the validator.
+- **Queued unbond number**: The number of unbond requests to be claimed, and 0 means claim BNB and rewards from
+  all the unbond requests.
+
+The `Claim` transaction will return the delegated BNB and rewards to the delegator. Be noted, a delegator can only get
+the rewards after unbond. Before undelegation, the reward will be furthur staked to boost a delegator's income.
```
Comment (zjubfd) [inline docs/bcfusion/validators/staking.md:132]: Can we have a slash.md to explain the three kind of slash?

File: docs/bcfusion/validators/staking.md (hunk 1 above)
Comment (forcodedancing) [reply docs/bcfusion/validators/staking.md:10]: fixed

File: docs/bcfusion/validators/staking.md (hunk 2 above)
Comment (forcodedancing) [reply docs/bcfusion/validators/staking.md:44]: updated

File: docs/bcfusion/validators/staking.md (hunk 3 above)
Comment (forcodedancing) [reply docs/bcfusion/validators/staking.md:124]: fixed the typo

File: docs/bcfusion/validators/gov.md (hunk 4 above)
Comment (forcodedancing) [reply docs/bcfusion/validators/gov.md:32]: sure, removed

File: docs/bcfusion/validators/gov.md (hunk 5 above)
Comment (forcodedancing) [reply docs/bcfusion/validators/gov.md:42]: fixed

File: docs/bcfusion/validators/staking.md (hunk 9 above)
Comment (forcodedancing) [reply docs/bcfusion/validators/staking.md:7]: fixed

File: docs/bcfusion/validators/staking.md (hunk 10 above)
Comment (forcodedancing) [reply docs/bcfusion/validators/staking.md:8]: reword-ed the sentence

File: docs/bcfusion/validators/staking.md (hunk 11 above)
Comment (forcodedancing) [reply docs/bcfusion/validators/staking.md:18]: fixed

File: docs/bcfusion/validators/staking.md (hunk 13 above)
Comment (forcodedancing) [reply docs/bcfusion/validators/staking.md:40]: updated

File: docs/bcfusion/validators/staking.md (hunk 14 above)
Comment (forcodedancing) [reply docs/bcfusion/validators/staking.md:43]: updated

File: docs/bcfusion/validators/staking.md (hunk 15 above)
Comment (forcodedancing) [reply docs/bcfusion/validators/staking.md:107]: added

File: docs/bcfusion/validators/gov.md (hunk 6 above)
Comment (forcodedancing) [reply docs/bcfusion/validators/gov.md:57]: In `execution` part, there is some introduction of this topic.

File: docs/bcfusion/validators/staking.md (hunk 16 above)
Comment (forcodedancing) [reply docs/bcfusion/validators/staking.md:119]: added in delegation part

File: docs/bcfusion/validators/staking.md (hunk 17 above)
Comment (forcodedancing) [reply docs/bcfusion/validators/staking.md:142]: added

File: docs/bcfusion/validators/gov.md (hunk 4 above)
Comment (zjubfd) [reply docs/bcfusion/validators/gov.md]: We change the logic to: one user can only have one active/pending proposal now.

File: docs/bcfusion/validators/staking.md (hunk 15 above)
Comment (zjubfd) [reply docs/bcfusion/validators/staking.md]: it is not 2%, it is 0.02%

File: docs/bcfusion/validators/slash.md [hunk 18]
```diff
@@ -0,0 +1,52 @@
+# Slash
+
+The BNB smart chain (BSC) is a blockchain network that aims to provide fast, secure, and reliable transactions. To
+achieve this, the BSC relies on a set of validators who are responsible for producing and validating blocks. Validators
+stake their BNB tokens to participate in the network and earn rewards.
+
+However, validators also face the risk of losing their stakes if they behave in ways that could harm the network's
+integrity and reliability. This is where the slashing mechanism comes in. The slashing mechanism is a set of rules and
+functions implemented in the `StakeHub` contract
```
Comment (zjubfd) [inline docs/bcfusion/validators/slash.md:9]: I think it is implement in the `SlashIndicator.sol`

File: docs/bcfusion/validators/slash.md [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+The slashing mechanism covers three types of offenses: downtime, double signing, and malicious voting. Each offense has
+a different severity and penalty, depending on the impact it has on the network. In this document, we will explain the
+slashing conditions and mechanisms for each offense in detail.
+
+## Downtime Slash
+
+Validators are expected to maintain high availability to ensure the network's smooth operation. Validators failing to
+meet these uptime requirements are subject to slashing.
+
+- The `StakeHub` contract tracks validator uptime by measuring the number of blocks they sign within a certain window.
```
Comment (zjubfd) [inline docs/bcfusion/validators/slash.md:23]: Not stakehub

File: docs/bcfusion/validators/slash.md [hunk 20]
```diff
@@ -0,0 +1,52 @@
...
+- If a validator fails to sign at least a minimum percentage of blocks within the window, they are considered offline.
```
Comment (zjubfd) [inline docs/bcfusion/validators/slash.md:24]: I think not percentage, but fixed number

File: docs/bcfusion/validators/slash.md [hunk 21]
```diff
@@ -0,0 +1,52 @@
...
+  have their stakes slashed by a predetermined amount.
+- The penalty for downtime is less severe than for double signing but is designed to encourage validators to maintain
+  the necessary infrastructure for high availability.
+
+## Double Sign Slash
+
+A critical offense within the BSC network is when a validator signs two different blocks at the same height. Such
+actions can lead to network forks, undermining the blockchain's security and consistency.
+
+- The `StakeHub` contract monitors for double signing by validators by comparing the signatures and hashes of the blocks
```
Comment (zjubfd) [inline docs/bcfusion/validators/slash.md:35]: not stakehub

File: docs/bcfusion/validators/slash.md [hunk 22]
```diff
@@ -0,0 +1,52 @@
...
+Validators are expected to maintain high availability to ensure the network's smooth operation. Validators failing to
+meet these uptime requirements are subject to slashing.
+
+- The `StakeHub` contract tracks validator uptime by measuring the number of blocks they sign within a certain window.
+- If a validator fails to sign at least a minimum percentage of blocks within the window, they are considered offline.
+- If a validator is offline for an extended period, beyond the acceptable thresholds defined in the contract, they will
+  have their stakes slashed by a predetermined amount.
+- The penalty for downtime is less severe than for double signing but is designed to encourage validators to maintain
+  the necessary infrastructure for high availability.
+
```
Comment (zjubfd) [inline docs/bcfusion/validators/slash.md:29]: Please paste the default slash params, like how many BNB slash and how long in jail

File: docs/bcfusion/validators/slash.md (hunk 21 above)
Comment (zjubfd) [reply docs/bcfusion/validators/slash.md:35]: Please paste the default slash params, like how many BNB slash and how long in jail

File: docs/bcfusion/validators/slash.md [hunk 23]
```diff
@@ -0,0 +1,52 @@
...
+A critical offense within the BSC network is when a validator signs two different blocks at the same height. Such
+actions can lead to network forks, undermining the blockchain's security and consistency.
+
+- The `StakeHub` contract monitors for double signing by validators by comparing the signatures and hashes of the blocks
+  they sign.
+- If a validator is caught double signing, the contract executes a slashing function, reducing the validator's stake and
+  moving them to a "jailed" state, preventing them from participating in consensus until manual intervention is taken.
+- The penalty for double signing is severe, with a significant portion of the validator's stake being slashed. This acts
+  as a strong deterrent against such malicious behavior.
```
Comment (zjubfd) [inline docs/bcfusion/validators/slash.md:40]: Please also add how to submit evidence to the contract.

File: docs/bcfusion/validators/slash.md [hunk 24]
```diff
@@ -0,0 +1,52 @@
...
+
+## Malicious Vote Slash
+
+Validators voting maliciously, such as voting for invalid transactions or blocks, pose a threat to the network's
+integrity. The BSC penalizes such actions to maintain the network's security and trustworthiness.
+
+- The `StakeHub` contract is responsible for monitoring the votes cast by validators and verifying their validity and
+  consistency.
+- Votes identified as malicious trigger the slashing mechanism, reducing the offender's stake and potentially jailing
+  the validator, depending on the severity of the offense.
```
Comment (zjubfd) [inline docs/bcfusion/validators/slash.md:50]: Is this generate by AI? not good

File: docs/bcfusion/validators/slash.md [hunk 25]
```diff
@@ -0,0 +1,52 @@
...
+- The `StakeHub` contract monitors for double signing by validators by comparing the signatures and hashes of the blocks
+  they sign.
+- If a validator is caught double signing, the contract executes a slashing function, reducing the validator's stake and
+  moving them to a "jailed" state, preventing them from participating in consensus until manual intervention is taken.
+- The penalty for double signing is severe, with a significant portion of the validator's stake being slashed. This acts
+  as a strong deterrent against such malicious behavior.
+
+## Malicious Vote Slash
+
+Validators voting maliciously, such as voting for invalid transactions or blocks, pose a threat to the network's
```
Comment (zjubfd) [inline docs/bcfusion/validators/slash.md:44]: Please refer read to the fast finality BEP

File: docs/bcfusion/validators/gov.md (hunk 4 above)
Comment (forcodedancing) [reply docs/bcfusion/validators/gov.md]: updated.

File: docs/bcfusion/validators/staking.md (hunk 15 above)
Comment (forcodedancing) [reply docs/bcfusion/validators/staking.md]: fixed

File: docs/bcfusion/validators/slash.md (hunk 24 above)
Comment (forcodedancing) [reply docs/bcfusion/validators/slash.md:50]: had ask AI to refine my writings.

File: docs/bcfusion/validators/slash.md (hunk 22 above)
Comment (forcodedancing) [reply docs/bcfusion/validators/slash.md:27]: added

File: docs/bcfusion/validators/slash.md (hunk 21 above)
Comment (forcodedancing) [reply docs/bcfusion/validators/slash.md]: added

File: docs/bcfusion/validators/slash.md (hunk 19 above)
Comment (forcodedancing) [reply docs/bcfusion/validators/slash.md]: fixed

File: docs/bcfusion/validators/slash.md (hunk 18 above)
Comment (forcodedancing) [reply docs/bcfusion/validators/slash.md]: updated

File: docs/bcfusion/validators/slash.md (hunk 23 above)
Comment (forcodedancing) [reply docs/bcfusion/validators/slash.md]: added

**Review by zjubfd — APPROVED** [review]

**Pull request merged**